### PR TITLE
Implements version comparator for rpm version tags

### PR DESCRIPTION
**Pull request by apozsuse**
Investigating this issue, and as it seems not easy to reuse the comparison code from `rpm` itself as it has problems with virtual environments, I saw how [Uyuni project deals with this issue](https://github.com/uyuni-project/obs-to-maven/blob/master/obs_maven/rpm.py) and adapted the code for our needs taking as base Sean's branch.

The `compare_rpm_labels` functions expects tuples with:
- epoch time
- version
- release

As our API for comparison has the `version` and `release` fields joined, this solution creates on the fly the tuples with:
- Similar epoch and release fields so they don't interfere in the comparison.
- All the available version tag as version.

### CODE REVIEW

File: obs_img_utils/rpm.py
Comment (smarlowucf) [inline obs_img_utils/rpm.py:1]: If this is mostly from Yuyuni I don't know exactly how we are supposed to handle the copyright and license properly. 🤔

**Review by smarlowucf — APPROVED** [review]
LGTM, thanks! The only question I have is how to properly handle license and copyright.

**Review by schaefi — APPROVED** [review]
I like it. The class basically implements the C interface of librpm, this should be safe for any package comparison we need :+1: 

Like with Sean I'm not sure about the correctness of the GPL licensing of the rpm namespace

**Comment by apozsuse** [comment]
As I was not sure about the Licensing either (as the code in Uyuni was apparently pasted from an StackOverflow answer, so it would be unclear), reimplemented the C version of the compare function in my own (took as reference [this other implementation with GPLv3 license](https://github.com/ihiji/version_utils/blob/development/version_utils/rpm.py):

- The function from RPM lib is [here](https://github.com/rpm-software-management/rpm/blob/9e4caf0fc536d1244b298abd9dc4c535b6560d69/rpmio/rpmvercmp.c)
- Some documentation about complex versioning (with `^` and `~` character meaning) can be found [here](https://docs.fedoraproject.org/en-US/packaging-guidelines/Versioning/#_complex_versioning)
- Test suite is [rpm's](https://github.com/rpm-software-management/rpm/blob/e03674aae18d15f5073e95adc5443c3b2c5ab7c3/tests/rpmvercmp.at#L4)

**Review by smarlowucf — APPROVED** [review]

**Pull request merged**